File: python/raindrops/raindrops.py
"""Raindrop sounds.

Convert a number into a string that contains raindrop sounds
corresponding to certain potential factors.
"""
RAINDROPS = {3: 'Pling', 5: 'Plang', 7: 'Plong'}


def convert(number: int) -> str:
    """Convert given number to raindrop sounds.

    If has raindrop factors, retrun their concatinated sounds
    string. Otherwise, retrun the number's didgits atring.

    :param number: the number to convert
    :retern: raindrop sounds, or digits of the number
    """
    output = ''
    for factor in RAINDROPS:
        if number % factor == 0:
            output += RAINDROPS[factor]

    if not output:
        output = str(number)

    return output

File: javascript/raindrops/raindrops.js
/**
 * Raindrops.
 * Convert a number into a string that contains raindrop sounds 
 * corresponding to certain potential factors. If now factors
 * found, result should be the digits of the number.
 */

/**
 * Orderd raindrops information.
 */
const RAINDROPS = [
  { factor: 3, sound: 'Pling' },
  { factor: 5, sound: 'Plang' },
  { factor: 7, sound: 'Plong' }
];

/**
 * Convert number to raindrops sound string according to number's factors.
 * @param {number} number - Input number.
 * @returns {string} The raindrops sound string of given number.
 */
export const convert = (number) => {
  let result = '';
  for (const raindrop of RAINDROPS) {
    if (number % raindrop.factor === 0)
      result += raindrop.sound;
  }

  return result || number.toString();
};
